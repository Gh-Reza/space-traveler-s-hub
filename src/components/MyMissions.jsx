import { useSelector } from 'react-redux';

const MyMissions = () => {
  const missions = useSelector((store) => store.missions.missions);
  const myMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div className="my-missions">
      <h2 className="heading">My Missions</h2>
      {myMissions.map((myMission) => (
        <p className="my-mission" key={myMission.id}>{myMission.mission_name}</p>
      ))}
    </div>
  );
};

export default MyMissions;
