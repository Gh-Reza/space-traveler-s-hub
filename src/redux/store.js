import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsslice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
}, applyMiddleware);

export default store;
