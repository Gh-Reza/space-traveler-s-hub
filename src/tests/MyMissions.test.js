import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import MyMissions from '../components/MyMissions';

describe('My Profile My Missions Component unit test', () => {
  test('MyMissions Renders Properly ', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MyMissions/>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});