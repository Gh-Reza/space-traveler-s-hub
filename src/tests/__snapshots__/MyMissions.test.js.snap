// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Profile My Missions Component unit test MyMissions Renders Properly  1`] = `
<div
  className="my-missions"
>
  <h2
    className="heading"
  >
    My Missions
  </h2>
</div>
`;
